### PR TITLE
Suggest on closure args count mismatching with pipe span

### LINKED ISSUE #55891: Suggestions in errors ignore `move` before closure arguments
Source: https://travis-ci.org/rust-lang/cargo/jobs/453802414#L938-L949

```
error[E0593]: closure is expected to take 1 argument, but it takes 0 arguments
   --> src/cargo/core/compiler/job_queue.rs:412:23
    |
399 |         let doit = move || {
    |                    ------- takes 0 arguments
...
412 |                 scope.spawn(doit);
    |                       ^^^^^ expected closure that takes 1 argument
help: consider changing the closure to take and ignore the expected argument
    |
399 |         let doit = |_| {
    |                    ^^^
```

The suggestion is to use: `let doit = |_| {`
But I think it should be: `let doit = move |_| {`

We should probably respect other modifiers like `static` and `async`, too.

### PR DESCRIPTION
**Pull request by csmoe**
Closes #55891 
r? @estebank

### CODE REVIEW

File: src/test/ui/mismatched_types/closure-arg-count.stderr [hunk 1]
```diff
@@ -60,8 +60,26 @@ help: consider changing the closure to take and ignore the expected argument
...
+   |     expected closure that takes 1 argument
+   |
+note: required by `f`
+  --> $DIR/closure-arg-count.rs:13:1
+   |
+LL | fn f<F: Fn<usize>>(_: F) {}
+   | ^^^^^^^^^^^^^^^^^^^^^^^^
+help: consider changing the closure to take and ignore the expected argument
+   |
+LL |     f(move|_| panic!());
```
Comment (csmoe) [inline src/test/ui/mismatched_types/closure-arg-count.stderr]: oops, whitespace between prefix and pipe should be reserved, fix soon.

**Comment by estebank** [comment]
r=me after fixing the whitespace issue.

**Comment by csmoe** [comment]
r=@estebank

**Comment by estebank** [comment]
@bors r+ rollup

@csmoe you need to mention bors for it to work: `@bors r=@estebank`

**Pull request merged**